src/sage/rings/function_field/drinfeld_modules/finite_drinfeld_module.py

```diff
@@ -183,46 +257,62 @@ def frobenius_endomorphism(self):
         deg = self.base_over_constants_field().degree_over()
         return self._Hom_(self, category=self.category())(t**deg)
 
-    def frobenius_charpoly(self, var='X'):
+    def frobenius_charpoly(self, var='X', algorithm='crystalline'):
```

Comment:
Shouldn't `'motive'` be an available option for the `algorithm` flag? This is implemented in `DrinfeldModuleMorphism`.

Reply:
I assumed that would get added when you and Xavier implement the other algorithms specialized for the frobenius case, but I can add a wrapper for the morphism method if you'd like.

Reply:
I'm actually fine with both solutions: adding it here or in a separate PR.
However, I think that if you add it, you should document it a bit.